### PR TITLE
Trim return value of prometheusHttpPrefix and alertManagerHttpPrefix

### LINKED ISSUE #3086: Invalid ingress nginx config due to newline in prometheusHttpPrefix Helm named templates
#### Describe the bug
Return value from named templates mimir.prometheusHttpPrefix is not trimmed. This will result in an invalid ingress Nginx config generated when `api.prometheus_http_prefix` is set.

#### To Reproduce

Prepare simple mimir-distributed `custom.yaml`.

```
mimir:
  config: |
    activity_tracker:
        filepath: /data/active-query-tracker
    api:
      prometheus_http_prefix: /prometheus
      alertmanager_http_prefix: /alertmanager
```

Install mimir-distributed using above helm values.

That will create the following invalid nginx config:
```
...
   # Rest of 
/prometheus goes to the query frontend
    location 
/prometheus {
      proxy_pass      http://cortex-query-frontend.cortex.svc.cluster.local:8080$request_uri;
    }
...
```

That config will make nginx ingress pod to fail starting with the following error:
```
2022/09/29 11:15:00 [emerg] 1#1: unknown directive "/prometheus" in /etc/nginx/nginx.conf:84
nginx: [emerg] unknown directive "/prometheus" in /etc/nginx/nginx.conf:84
```

#### Expected behavior

The nginx config should be generated properly as follow.

```
   # Rest of /prometheus goes to the query frontend
    location 
/prometheus {
      proxy_pass      http://cortex-query-frontend.cortex.svc.cluster.local:8080$request_uri;
    }
```

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: helm

#### Additional Context

We can fix this by removing that templated comment `# Rest of /prometheus goes to the query frontend` or to just trim the prometheusHttpPrefix named templates. alertmanagerHttpPrefix template is not giving any issue since it is not used in any comment in nginx config. However, it is good to trim that as well.

### PR DESCRIPTION
**Pull request by lamida**
#### Which issue(s) this PR fixes or relates to

Fixes #3086

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by dimitarvdimitrov — COMMENTED** [review]
Thanks for fixing this! Should we add a changelog entry too?

**Comment by lamida** [comment]
> Thanks for fixing this! Should we add a changelog entry too?

Yes, this is still a draft. Will add everything required. :)

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Monitoring: Add additional flags to conditionally enable log / metric scraping. #2936
 * [ENHANCEMENT] Add podAntiAffinity to sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml). #2906
 * [ENHANCEMENT] Update agent-operator subchart to `0.2.5`. #3009
+- [BUGFIX] Fix Invalid ingress nginx config due to newline in prometheusHttpPrefix Helm named templates. #3087
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:38]:
```suggestion
* [BUGFIX] Fix Invalid ingress nginx config due to newline in prometheusHttpPrefix Helm named templates. #3087
```

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM

**Comment by pracucci** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Review by krajorama — CHANGES_REQUESTED** [review]
Please rebase as 3.2.0 changelog was cut already, otherwise ok

**Comment by lamida** [comment]
> Please rebase as 3.2.0 changelog was cut already, otherwise ok

@krajorama reabase is done.

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 2]
```diff
@@ -86,6 +86,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Add podAntiAffinity to sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml). #2906
 * [ENHANCEMENT] Add ability to configure and run mimir-continuous-test. #3117
 * [BUGFIX] Fix wrong label selector in ingester anti affinity rules in the sizing plans. #2906
+* [BUGFIX] Fix Invalid ingress nginx config due to newline in prometheusHttpPrefix Helm named templates. #3088
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:89]: should be moved to main/unreleased as it will not be part of 3.2 currently

**Review by krajorama — APPROVED** [review]

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [BUGFIX] Fix an issue that caused metamonitoring secrets to be created incorrectly #3170
 * [BUGFIX] Nginx: fixed `imagePullSecret` value reference inconsistency. #3208
 * [BUGFIX] Move the activity tracker log from /data to /active-query-tracker to remove ignore log messages. #3169
+* [BUGFIX] Fix Invalid ingress nginx config due to newline in prometheusHttpPrefix Helm named templates. #3088
```
Comment (replay) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:35]: should this be #3087 ?

Comment (pracucci) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:35]: Yes. Thanks for noticing it!

Comment (lamida) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:35]: Thank you for catching this :(. I might accidentally increment the number during editing. Will be more careful next time.

**Pull request merged**